pkg/oc/admin/migrate/legacyhpa/hpa.go

```diff
@@ -0,0 +1,209 @@
+package legacyhpa
+
+import (
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	autoscaling "k8s.io/kubernetes/pkg/apis/autoscaling"
+	autoscalingclient "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset/typed/autoscaling/internalversion"
+	"k8s.io/kubernetes/pkg/kubectl/cmd/templates"
+	kcmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/kubectl/resource"
+
+	"github.com/openshift/origin/pkg/oc/admin/migrate"
+	"github.com/openshift/origin/pkg/oc/cli/util/clientcmd"
+)
+
+var (
+	defaultMigrations = map[versionKind]versionKind{
+		// legacy oapi group
+		{"v1", "DeploymentConfig"}: {"apps.openshift.io/v1", "DeploymentConfig"},
+
+		// webconsole shenaniganry
+		{"extensions/v1beta1", "DeploymentConfig"}:      {"apps.openshift.io/v1", "DeploymentConfig"},
```

Comment:
Say what? That should never happen.

Reply:
The web console was trying to target the `scale` subresource in the `extensions/v1beta1` group (I think based on an earlier HPA API version), but the subresource was getting dropped. So it unfortunately got set as `extensions/v1beta1.DeploymentConfig`.

Reply:
😭